File: JavaDroidDemo/app/src/main/java/com/demo/reynzdev/javadroiddemo/MainActivity.java
package com.demo.reynzdev.javadroiddemo;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.net.Uri;
import android.os.BatteryManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button emailButton = findViewById(R.id.emailBtn);
        Button batButton = findViewById(R.id.batBtn);
        Button callButton = findViewById(R.id.callBtn);

        emailButton.setOnClickListener(this);
        batButton.setOnClickListener(this);
        callButton.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.emailBtn:
                sendEmail();
                break;
            case R.id.callBtn:
                call();
                break;
            case R.id.batBtn:
                checkBatteryStatus();
                break;
            default:
        }
    }

    private void call() {
        Intent intent = new Intent(Intent.ACTION_DIAL);
        intent.setData(Uri.parse("tel:9988776655"));
        startActivity(intent);
    }

    private void sendEmail() {
        Intent emailIntent = new Intent(Intent.ACTION_SEND);
        emailIntent.setType("text/plain");
        emailIntent.putExtra(Intent.EXTRA_EMAIL,"johndoe@example.com");
        emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Hello World!");
        emailIntent.putExtra(Intent.EXTRA_TEXT,"Hi! I am sending you a test email");

        try {
            startActivity(Intent.createChooser(emailIntent, "Send e-mail"));
        } catch (android.content.ActivityNotFoundException ex) {
            Toast.makeText(MainActivity.this, "There is no email client installed.",
                    Toast.LENGTH_SHORT).show();
        }
    }

    private void checkBatteryStatus() {
        IntentFilter intentFilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
        MyBroadcastReceiver broadcastReceiver = new MyBroadcastReceiver(this);
        registerReceiver(broadcastReceiver,intentFilter);
    }

    public class MyBroadcastReceiver extends BroadcastReceiver {

        MainActivity mMainActivity;

        public MyBroadcastReceiver(MainActivity mainActivity) {
            mMainActivity = mainActivity;
        }

        @Override
        public void onReceive(Context context, Intent intent) {
            // status
            int status = intent.getIntExtra(BatteryManager.EXTRA_STATUS, -1);

            switch (status) {
                case 1:         // Unknown
                case 3:         // Discharging
                case 4:         // NotCharging
                case 5:         // Full
                    int level = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1);
                    int scale = intent.getIntExtra(BatteryManager.EXTRA_SCALE, -1);
                    float batPercentage = (level/(float)scale) * 100;
                    String batLevel = "Battery remaining " + (int)batPercentage + "%";
                    Toast.makeText(mMainActivity,batLevel,Toast.LENGTH_SHORT).show();
                    break;
                case 2:
                    Toast.makeText(mMainActivity,"Charging",Toast.LENGTH_SHORT).show();
                    break;
                default:
                    break;
            }
        }
    }
}

File: XamarinDemo/XamarinAndroidDemo/MainActivity.cs
using Android.App;
using Android.OS;
using Android.Support.V7.App;
using Android.Runtime;
using Android.Widget;
using Android.Content;
using Android.Net;

namespace XamarinAndroidDemo
{
    [Activity(Label = "@string/app_name", Theme = "@style/AppTheme", MainLauncher = true)]
    public class MainActivity : AppCompatActivity
    {
        IntentFilter intentFilter;

        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            SetContentView(Resource.Layout.activity_main);

            var emaiBtn = FindViewById<Button>(Resource.Id.emailBtn);
            var callBtn = FindViewById<Button>(Resource.Id.callBtn);
            var batBtn = FindViewById<Button>(Resource.Id.batBtn);

            emaiBtn.Click += SendEmail;
            callBtn.Click += Call;
            batBtn.Click += CheckBatteryStatus;
        }

        private void Call(object sender, System.EventArgs e)
        {
            Intent intent = new Intent(Intent.ActionDial);
            intent.SetData(Uri.Parse("tel:9988776655"));
            StartActivity(intent);
        }

        private void SendEmail(object sender, System.EventArgs e)
        {
            Intent emailIntent = new Intent(Intent.ActionSend);
            emailIntent.SetType("text/plain");
            emailIntent.PutExtra(Intent.ExtraEmail, "johndoe@example.com");
            emailIntent.PutExtra(Intent.ExtraSubject, "Hello World!");
            emailIntent.PutExtra(Intent.ExtraText, "Hi! I am sending you a test email");

            try
            {
                StartActivity(Intent.CreateChooser(emailIntent, "Send e-mail"));
            }
            catch (ActivityNotFoundException ex)
            {
                Toast.MakeText(this, "There is no email client installed.",
                    ToastLength.Short).Show();
            }
        }

        private void CheckBatteryStatus(object sender, System.EventArgs e)
        {
            intentFilter = new IntentFilter(Intent.ActionBatteryChanged);
            MyBroadcastReceiver broadcastReceiver = new MyBroadcastReceiver(this);
            RegisterReceiver(broadcastReceiver, intentFilter);
        }
    }

    public class MyBroadcastReceiver : BroadcastReceiver
    {
        private MainActivity mainActivity;

        public MyBroadcastReceiver(MainActivity mainActivity)
        {
            this.mainActivity = mainActivity;
        }

        public override void OnReceive(Context context, Intent intent)
        {
            // status
            int status = intent.GetIntExtra(BatteryManager.ExtraStatus, -1);

            switch (status)
            {
                case 1:         // Unknown
                case 3:         // Discharging
                case 4:         // NotCharging
                case 5:         // Full
                    int level = intent.GetIntExtra(BatteryManager.ExtraLevel, -1);
                    int scale = intent.GetIntExtra(BatteryManager.ExtraScale, -1);
                    float batteryPct = (level / (float)scale) * 100;
                    var batLevel = $"Battery remaining {batteryPct}%";
                    Toast.MakeText(mainActivity, batLevel, ToastLength.Short).Show();
                    break;
                case 2:         // Charging 
                    Toast.MakeText(mainActivity, "Charging", ToastLength.Short).Show();
                    break;
                default:
                    break;

            }
        }
    }
}